capture.js

```diff
@@ -1,26 +1,38 @@
-var capture = function(){
-    var replace_all = function(str, find, replace) { return str.replace(new RegExp(find, 'g'), replace); };
-    var esc = function (text) { return replace_all(replace_all(replace_all(encodeURIComponent(text),"[(]",escape("(")),"[)]",escape(")")),"[']",escape("'")); };
-    var selection = window.getSelection().toString();
-
-    var uri = 'org-protocol://';
-    if (selection != "")
-    {
-	uri += 'capture:/p/';
-    }
-    else
-    {
-	uri += 'capture:/L/'
-    };
+function replace_all(str, find, replace) {
+    return str.replace(new RegExp(find, 'g'), replace);
+}
 
-    uri += encodeURIComponent(location.href) + '/' +
-	esc(document.title) ;
+function escapeIt(text) {
+    return replace_all(replace_all(replace_all(encodeURIComponent(text), "[(]", escape("(")),
+                                   "[)]", escape(")")),
+                       "[']" ,escape("'"));
+}
 
-    if (selection != "") { uri += '/' + esc(selection); };
+function createCaptureURL(template, title, url, selection, oldStyle) {
+    if (oldStyle == true)
+        return "org-protocol://capture:/"+template+'/'+url+'/'+title + ((selection === '') ? '' : ('/' + selection));
+    else
+        return "org-protocol://capture?template="+template+'&url='+url+'&title='+title+((selection === '') ? '' : ('&body=' + selection));
```

Comment:
Since you are using a regular URL syntax here, the selection should be escaped using encodeURIComponent as well.